test/e2e/mcd_test.go

```diff
@@ -92,57 +86,70 @@ func TestMCDeployed(t *testing.T) {
 			},
 			Storage: ignv2_2types.Storage{
 				Files: []ignv2_2types.File{
-					createMCFile("/etc/mytestconf", "data:,test", 420),
+					createIgnFile(filename, "data:,"+data, 420),
 				},
 			},
 		},
 	}
+	return mcadd
+}
 
-	// create the dummy MC now
-	_, err = mcClient.MachineconfigurationV1().MachineConfigs().Create(mcadd)
+func TestMCDeployed(t *testing.T) {
+	cb, err := common.NewClientBuilder("")
 	if err != nil {
-		t.Errorf("failed to create machine config %v", err)
+		t.Errorf("%#v", err)
 	}
+	mcClient := cb.MachineConfigClientOrDie("mc-file-add")
+	k := cb.KubeClientOrDie("mc-file-add")
 
-	// grab the latest worker- MC
-	var newMCName string
-	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
-		mcp, err := mcClient.MachineconfigurationV1().MachineConfigPools().Get("worker", metav1.GetOptions{})
-		if err != nil {
-			return false, err
-		}
-		for _, mc := range mcp.Status.Configuration.Source {
-			if mc.Name == mcName {
-				newMCName = mcp.Status.Configuration.Name
-				return true, nil
-			}
-		}
-		return false, nil
-	}); err != nil {
-		t.Errorf("machine config hasn't been picked by the pool: %v", err)
-	}
+	for i := 0; i < 10; i++ {
+		mcadd := createMCToAddFile(fmt.Sprintf("/etc/mytestconf%d", i), "test")
 
-	visited := make(map[string]bool)
-	if err := wait.Poll(2*time.Second, 5*time.Minute, func() (bool, error) {
-		nodes, err := getNodesByRole(k, "worker")
+		// create the dummy MC now
+		_, err = mcClient.MachineconfigurationV1().MachineConfigs().Create(mcadd)
 		if err != nil {
-			return false, err
+			t.Errorf("failed to create machine config %v", err)
 		}
-		for _, node := range nodes {
-			if visited[node.Name] {
-				continue
+
+		// grab the latest worker- MC
+		var newMCName string
+		if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+			mcp, err := mcClient.MachineconfigurationV1().MachineConfigPools().Get("worker", metav1.GetOptions{})
+			if err != nil {
+				return false, err
 			}
-			if node.Annotations[constants.CurrentMachineConfigAnnotationKey] == newMCName {
-				visited[node.Name] = true
-				if len(visited) == len(nodes) {
+			for _, mc := range mcp.Status.Configuration.Source {
+				if mc.Name == mcadd.Name {
```

Comment:
I don't understand this loop.  It's going to be racy; the renderer could have just seen the first config and generated a new rendered version and be working on the full one when we drop in here.

I think the check needs to be that `set(sources) = set(mcadd)`.

Reply:
we won't create the second config unless the previous one is applied to all nodes, am I missing something?

Reply:
this check makes sure that the created MC is in the rendered, generated one that we're now going to check (and everything is serial as well)

Reply:
Oh right it's inside the loop.  OK.